Cabal/src/Distribution/Simple/Configure.hs

```diff
@@ -767,22 +768,15 @@ configure (pkg_descr0, pbi) cfg = do
             )
           return False
 
-  let compilerSupportsGhciLibs :: Bool
-      compilerSupportsGhciLibs =
-        case compilerId comp of
-          CompilerId GHC version
-            | version > mkVersion [9, 3] && windows ->
-                False
-          CompilerId GHC _ ->
-            True
-          CompilerId GHCJS _ ->
-            True
-          _ -> False
-        where
-          windows = case compPlatform of
-            Platform _ Windows -> True
-            Platform _ _ -> False
-
+  let linkerSupportsGhciLibs :: Bool
+      linkerSupportsGhciLibs =
+        case lookupProgramByName "ld" programDb'' of
+          Nothing -> True -- NOTE: This may still fail if the linker does not support -r.
+          Just ld ->
+            case Map.lookup "Supports relocatable output" $ programProperties ld of
+              Just "YES" -> True
+              Just "NO" -> False
+              _other -> True -- NOTE: This may still fail if the linker does not support -r.
```

Comment:
Sorry to be a PITA :joy:

What this function does is checking whether we had previously detected that linker **does not** support `-r`. Am I correct? We use this information to ignore `enable-library-for-ghci` further down. I think this should be more explicit.

How about we
1. Rename the function to not mention ghci but to mention the linker property (relocatable outputs? is it always `-r`?)
2. Change the function to return Maybe Bool (as in Yes/No/I don't know).
3. Delegate the `fromMaybe True` to `withGHCiLib_`

Does this sound fair?

Reply:
Done.